camelot/view/controls/qml_progress_dialog.py

```diff
@@ -0,0 +1,447 @@
+#        notice, this list of conditions and the following disclaimer.
+#      * Redistributions in binary form must reproduce the above copyright
+#        notice, this list of conditions and the following disclaimer in the
+#        documentation and/or other materials provided with the distribution.
+#      * Neither the name of Conceptive Engineering nor the
+#        names of its contributors may be used to endorse or promote products
+#        derived from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+#  ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+#  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+#  DISCLAIMED. IN NO EVENT SHALL <COPYRIGHT HOLDER> BE LIABLE FOR ANY
+#  DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+#  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+#  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+#  ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+#  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+#  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+#  ============================================================================
+
+"""Functions and classes to use a progress dialog in combination with
+a model thread"""
+
+import logging
+
+from camelot.core.utils import ugettext, ugettext_lazy
+from camelot.view.art import FontIcon
+
+import six
+
+from ...core.qt import QtModel, QtCore, QtWidgets, Qt, QtQml, py_to_variant, is_deleted
+
+LOGGER = logging.getLogger( 'camelot.view.controls.qml_progress_dialog' )
+#LOGGER.setLevel(logging.DEBUG)
+
+class ProgressState:
+    def __init__(self):
+        self.label = ''
+        self.title = ugettext('Please wait')
+        self.event_loop = None
+        self.details = []
+        self.reset()
+
+    def reset(self):
+        self.minimum = 0
+        self.maximum = 100
+        self.value = 0
+        self.setValue_called = False
+        self.auto_close = True
+        self.auto_reset = True
+        self.hidden = False
+        self.ok_hidden = True
+        self.cancel_hidden = False
+
+
+class QmlProgressDialog(QtCore.QObject):
+    """
+A Progress Dialog, used during the :meth:`gui_run` of an action.
+
+.. image:: /_static/controls/progress_dialog.png
+    """
+    # FIXME: ^ update image
+
+    canceled = QtCore.qt_signal()
+
+    def __init__(self, quick_view):
+        super().__init__(parent=quick_view)
+        self.setObjectName('progress_dialog')
+        # create QML component
+        engine = quick_view.engine()
+        component = QtQml.QQmlComponent(engine)
+        component.loadUrl(QtCore.QUrl("qrc:/qml/progress_dialog/progress_dialog.qml"))
+        self._item = component.create()
```

Comment:
hier ook, geen QObjects in python aan elkaar verbinden, parent, findChild gebruiken

Reply:
done